packages/catalog-server/src/lib/firestore/firestore-repository.ts

```diff
@@ -255,73 +367,71 @@ export class FirestoreRepository implements Repository {
     version: string,
     tagName?: string
   ): Promise<CustomElement[]> {
-    const packageDocId = getPackageDocId(packageName);
-    const packageRef = db
-      .collection('packages')
-      .doc(packageDocId) as DocumentReference<PackageInfoData>;
-    const versionRef = packageRef.collection('versions').doc(version);
-    const customElementsRef = versionRef.collection('customElements');
+    const versionRef = this.getPackageVersionRef(packageName, version);
+    const customElementsRef = versionRef
+      .collection('customElements')
+      .withConverter(customElementConverter);
     let customElementsQuery:
       | CollectionReference<CustomElement>
-      | Query<CustomElement> = customElementsRef.withConverter(
-      customElementConverter
-    );
+      | Query<CustomElement> = customElementsRef;
     if (tagName !== undefined) {
       customElementsQuery = customElementsQuery.where('tagName', '==', tagName);
     }
     const customElementsResults = await customElementsQuery.get();
     return customElementsResults.docs.map((d) => d.data());
   }
-}
 
-/**
- * Returns a Firestore document ID based on the package name.
- */
-const getPackageDocId = (packageName: string) =>
-  packageName.replaceAll('/', '__');
+  getPackageRef(packageName: string) {
+    return db
+      .collection('packages' + this.namespace ? `-${this.namespace}` : '')
+      .doc(packageNameToId(packageName))
+      .withConverter(packageInfoConverter);
+  }
 
-const getPackageVersionRef = (packageName: string, version: string) => {
-  const packageDocId = getPackageDocId(packageName);
-  const packageRef = db.collection('packages').doc(packageDocId);
-  return packageRef.collection('versions').doc(version);
-};
+  getPackageVersionRef(packageName: string, version: string) {
+    return this.getPackageRef(packageName)
+      .collection('versions')
+      .doc(version)
+      .withConverter(packageVersionConverter);
+  }
+}
 
-/**
- * Generates a type representing a Firestore document from a GraphQL schema
- * type.
- *
- *  - Removes __typename
- *  - Date -> Timestamp
- *  - Removes specified collection fields
- *  - Transforms list of tuples to maps
- */
-type FirestoreType<
-  SchemaType,
-  MapFields extends {[k: string]: string},
-  Collections extends string
-> = {
-  [K in keyof SchemaType]: K extends '__typename'
-    ? never
-    : K extends Date
-    ? Timestamp
-    : K extends keyof MapFields
-    ? {
-        [key: string]: MapFields[K] extends string
-          ? SchemaType[K] extends ReadonlyArray<infer T>
-            ? Omit<T, MapFields[K]>
-            : never
-          : MapFields[K];
-      }
-    : K extends Collections
-    ? never
-    : SchemaType[K];
-};
+// /**
```

Comment:
Why comment this out?

Reply:
Oh. Not using it anymore, even though I do want to bring this back in some form. This is a type-only utility and I think I want to make an actual transformer function. I also was searching for a way to automate this from the GraphQL schema w/ field directives, but I think we'd have to write a plugin for that.

I'll delete it for now, but I'm likely to want to make a firestore/schema.ts module with something like this in it.